torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -1375,9 +1375,9 @@ def _are_inputs_layout_compatible(self, layouts: List[Layout]) -> bool:
             return False
         A_size = [int(i) for i in A_layout.size]
         B_size = [int(i) for i in B_layout.size]
-        K = max(A_size[-1], B_size[-2])
         M = A_size[-2]
         N = B_size[-1]
+        K = max(M, N)
```

Comment:
Wait this is still NOT fixed?

Reply:
That's a nasty bit of code, I cleaned it and changed the duplicated condition to actually make sense, let's see what the CI says!